src/pages/signin.tsx

```diff
@@ -0,0 +1,47 @@
+import { useEffect, type FC } from "react";
+import { Title } from "@mantine/core";
+import { signIn, useSession } from "next-auth/react";
+import { useRouter } from "next/router";
+import { IconUserPlus, IconBrandDiscord } from "@tabler/icons";
+
+const SignIn: FC = () => {
+  const { data: session } = useSession();
+  const router = useRouter();
+  const handleSignIn = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    signIn("discord", {
+      callbackUrl: "/notes",
+    });
+  };
+  useEffect(() => {
+    if (session) {
+      // eslint-disable-next-line @typescript-eslint/no-floating-promises
+      router.push("/notes");
```

Comment:
Please avoid using eslint-disable. I know it is annoying, but it could be solved 99% percent. This one just needs the keyword `void` in front of it. It basically says that the promise won't be used